xlators/performance/md-cache/src/md-cache.c

```diff
@@ -2915,6 +2948,31 @@ mdc_readdirp_cbk(call_frame_t *frame, void *cookie, xlator_t *this, int op_ret,
         }
     }
 
+    efd = fd_ctx_get(local->fd, this);
+    if ((efd != -1) && (size < 131072)) {
```

Comment:
Shouldn't the check be if efd != NULL?
(also, in the line above, I think the fd_ctx_get() should only be done if size is < 128K ?)

Reply:
Either the fd_ctx_get returns 0 or ctx as an unsigned value so need to check only the value should be non-zero. 
I will update it in the next patch.

Reply:
done